applications/StructuralMechanicsApplication/custom_processes/apply_multi_point_constraints_process.h

```diff
@@ -32,134 +29,206 @@
     template <int TDim>
-    void ApplyConstraints(ModelPart &master_model_part, ModelPart &slave_model_part)
+    void ApplyConstraints(
+        ModelPart &rMasterModelPart,
+        ModelPart &rSlaveModelPart
+        )
     {
-        BinBasedFastPointLocator<TDim> *p_point_locator = new BinBasedFastPointLocator<TDim>(master_model_part);
-        int numVars = m_parameters["variable_names"].size();
+        BinBasedFastPointLocator<TDim> *p_point_locator = new BinBasedFastPointLocator<TDim>(rMasterModelPart);
+        const SizeType num_vars = mParameters["variable_names"].size();
         // iterating over slave nodes to find the corresponding masters
-        const int n_slave_nodes = slave_model_part.Nodes().size();
+        const SizeType n_slave_nodes = rSlaveModelPart.Nodes().size();
         array_1d<double, TDim + 1> N; // This is only for triangular meshes
         const int max_results = 100;
         typename BinBasedFastPointLocator<TDim>::ResultContainerType results(max_results);
 
-        for (int i = 0; i < n_slave_nodes; i++)
-        {
-            ModelPart::NodesContainerType::iterator iparticle = slave_model_part.NodesBegin() + i;
-            Node<3>::Pointer p_slave_node = *(iparticle.base());
+        for (IndexType i = 0; i < n_slave_nodes; i++) {
+            ModelPart::NodesContainerType::iterator iparticle = rSlaveModelPart.NodesBegin() + i;
+            NodeType::Pointer p_slave_node = *(iparticle.base());
             typename BinBasedFastPointLocator<TDim>::ResultIteratorType result_begin = results.begin();
-            Element::Pointer pMasterElement;
-            bool is_found = false;
-            is_found = p_point_locator->FindPointOnMesh(p_slave_node->Coordinates(), N, pMasterElement, result_begin, max_results);
-            if (is_found == true)
-            {
-                for (int i = 0; i < numVars; i++)
-                {
-                    std::string varName = m_parameters["variable_names"][i].GetString();
-                    Geometry<Node<3>> &geom = pMasterElement->GetGeometry();
-                    for (unsigned int i = 0; i < geom.size(); i++)
-                    {
-                        if (KratosComponents<Variable<double>>::Has(varName))
-                        { //case of double variable
-                            VariableType rVar = KratosComponents<Variable<double>>::Get(m_parameters["variable_names"][i].GetString());
+            Element::Pointer p_master_element;
+            const bool is_found = p_point_locator->FindPointOnMesh(p_slave_node->Coordinates(), N, p_master_element, result_begin, max_results);
+            if (is_found == true) {
+                for (IndexType i = 0; i < num_vars; i++) {
+                    std::string varName = mParameters["variable_names"][i].GetString();
+                    Geometry<NodeType> &geom = p_master_element->GetGeometry();
+                    for (IndexType i = 0; i < geom.size(); i++) {
```

Comment:
again i??

Reply:
Not my original code...

Reply:
I will change this in a commit to this branch, and we can merge this together with this change. But I am surprised that it did not have any problem in the cases I used.